main.go

```diff
@@ -3,29 +3,97 @@
 package main
 
 import (
+	"context"
 	"errors"
 	"log"
+	"net"
+	"net/http"
 	"os"
 	"os/signal"
 	"syscall"
+	"time"
 
 	server "github.com/couchbase-examples/wasmcloud-provider-couchbase/bindings"
 	"github.com/couchbase/gocb/v2"
 	"github.com/wasmCloud/provider-sdk-go"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
 )
 
+func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodGet {
+		w.WriteHeader(http.StatusOK)
+		log.Default().Println("Health check successful")
+		// span.AddEvent("Health check successful")
+	} else {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		log.Default().Println("Health check failed")
+		// span.AddEvent("Health check failed")
+	}
+}
+
 func main() {
 	if err := run(); err != nil {
 		log.Fatal(err)
 	}
 }
 
+func newHTTPHandler() http.Handler {
+	mux := http.NewServeMux()
+	handleFunc := func(pattern string, handlerFunc func(http.ResponseWriter, *http.Request)) {
+		handler := otelhttp.WithRouteTag(pattern, http.HandlerFunc(handlerFunc))
+		mux.Handle(pattern, handler)
+	}
+
+	// Register handler for healthcheck.
+	handleFunc("/health", healthCheckHandler)
+
+	// Add HTTP instrumentation for the whole server.
+	handler := otelhttp.NewHandler(mux, "/")
+	return handler
+}
+
 func run() error {
+	// Handle SIGINT (CTRL+C) gracefully.
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stop()
+
+	// Set up OpenTelemetry.
+	otelShutdown, err := setupOTelSDK(ctx)
```

Comment:
In the future, I think it would be a good enhancement to not error out if OTEL can't be configured, in the case where we don't want to trace at all. This is fine as a future enhancement